theano/gpuarray/elemwise.py

```diff
@@ -2375,9 +2391,9 @@ def gpu_kernels(self, node, nodename):
                 const int threadCount = blockDim.x * blockDim.y * blockDim.z;
                 const int threadNum = threadIdx.z * blockDim.x * blockDim.y + threadIdx.y * blockDim.x + threadIdx.x;
                 extern __shared__ %(acc_type)s buf[];
-                %(acc_type)s myresult = %(reduce_init)s;
                 A = (const %(in_type)s *)(((char *)A)+offset_A);
                 Z = (%(out_type)s *)(((char *)Z)+offset_Z);
+                %(acc_type)s myresult = %(reduce_init)s;
```

Comment:
Can you also update the _k_init() method to have the same order? For consistency and in case of futur copy/paste. Idem around line 2030

Reply:
Done.